### ISSUE TITLE
Pm2 out of memory error

### ISSUE THREAD

**Issue by devhus** [nuxt/framework#8391]
### Environment

- Operating System: `Linux`
- Internal Host System: CPanel
- Node Version:     `v16.15.0`
- Nuxt Version:     `3.0.0-rc.12`
- Nitro Version:    `0.6.0`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `runtimeConfig`, `globalName`, `modules`, `build`, `vite`, `i18n`
- Runtime Modules:  `@nuxtjs/i18n@8.0.0-beta.1`, `nuxt-lodash@2.3.0`, `@pinia/nuxt@0.4.3`
- Build Modules:    `-`

### Reproduction

None

### Describe the bug

Running the Nuxt server using PM2 always gives a 500-page error and prints this error in logs,
Note that the server can run using other tools like CPanel Supervisor 

### Logs

```shell
51|yarn run start-prod  | [nuxt] [request error] [unhandled] [500] WebAssembly.instantiate(): Out of memory: wasm memory
51|yarn run start-prod  |   at node:internal/deps/cjs-module-lexer/dist/lexer:1:33593
51|yarn run start-prod  |   at async initCJSParse (node:internal/modules/esm/translators:64:5)
51|yarn run start-prod  |   at async ESMLoader.commonjsStrategy (node:internal/modules/esm/translators:156:18)
51|yarn run start-prod  | [nuxt] [request error] [unhandled] [500] WebAssembly.instantiate(): Out of memory: wasm memory
51|yarn run start-prod  |   at node:internal/deps/cjs-module-lexer/dist/lexer:1:33593
51|yarn run start-prod  |   at async initCJSParse (node:internal/modules/esm/translators:64:5)
51|yarn run start-prod  |   at async ESMLoader.commonjsStrategy (node:internal/modules/esm/translators:156:18)
51|yarn run start-prod  | [nuxt] [request error] [unhandled] [500] WebAssembly.instantiate(): Out of memory: wasm memory
51|yarn run start-prod  |   at node:internal/deps/cjs-module-lexer/dist/lexer:1:33593
51|yarn run start-prod  |   at async initCJSParse (node:internal/modules/esm/translators:64:5)
51|yarn run start-prod  |   at async ESMLoader.commonjsStrategy (node:internal/modules/esm/translators:156:18)
51|yarn run start-prod  | [nuxt] [request error] [unhandled] [500] WebAssembly.instantiate(): Out of memory: wasm memory
51|yarn run start-prod  |   at node:internal/deps/cjs-module-lexer/dist/lexer:1:33593
51|yarn run start-prod  |   at async initCJSParse (node:internal/modules/esm/translators:64:5)
51|yarn run start-prod  |   at async ESMLoader.commonjsStrategy (node:internal/modules/esm/translators:156:18)
51|yarn run start-prod  | [nuxt] [request error] [unhandled] [500] WebAssembly.instantiate(): Out of memory: wasm memory
51|yarn run start-prod  |   at node:internal/deps/cjs-module-lexer/dist/lexer:1:33593
51|yarn run start-prod  |   at async initCJSParse (node:internal/modules/esm/translators:64:5)
51|yarn run start-prod  |   at async ESMLoader.commonjsStrategy (node:internal/modules/esm/translators:156:18)
51|yarn run start-prod  | [nuxt] [request error] [unhandled] [500] WebAssembly.instantiate(): Out of memory: wasm memory
51|yarn run start-prod  |   at node:internal/deps/cjs-module-lexer/dist/lexer:1:33593
51|yarn run start-prod  |   at async initCJSParse (node:internal/modules/esm/translators:64:5)
51|yarn run start-prod  |   at async ESMLoader.commonjsStrategy (node:internal/modules/esm/translators:156:18)
```
```

**Comment by danielroe** [comment]
Would you provide a [_minimal_ reproduction](https://v3.nuxtjs.org/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Comment by devhus** [comment]
> Would you provide a [_minimal_ reproduction](https://v3.nuxtjs.org/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

No reproduction, the issue happens with a freshly created Nuxt project.
If it helps, that's how the PM2 instance starts:
- add ``.htaccess`` file to the project's root with the following script (Since the project is running on CPanel we need to proxy the subdomain to localhost port)
```
Options +FollowSymLinks -Indexes

IndexIgnore *

DirectoryIndex
<IfModule mod_rewrite.c>
RewriteEngine on
RewriteRule ^(.*)$ http://localhost:3005/$1 [P]
</IfModule>
```
- add ``.env`` file 
```
PORT=3005
HOST=0.0.0.0
```
- add ``"start-prod": "node -r dotenv/config .output/server/index.mjs dotenv_config_path=.env"`` to package.jsoin scripts
- $ pm2 start "yarn run start-prod"

**Comment by danielroe** [comment]
> No reproduction, the issue happens with a freshly created Nuxt project.

Your project does not appear to be a fresh Nuxt 3 project. You have added runtime modules, for example, and extra config.

Nor, indeed, can I reproduce it with a fresh Nuxt project. I would appreciate a minimal reproduction that you have verified produces the error on your own machine. 🙏

**Comment by devhus** [comment]
> > No reproduction, the issue happens with a freshly created Nuxt project.
> 
> Your project does not appear to be a fresh Nuxt 3 project. You have added runtime modules, for example, and extra config.
> 
> Nor, indeed, can I reproduce it with a fresh Nuxt project. I would appreciate a minimal reproduction that you have verified produces the error on your own machine. 🙏

Yes, the provided environment info is from the main project, But also installing a fresh Nuxt project on the same environment with the same method reproduces the issue with/without the runtime modules, and configs.

**Comment by devhus** [comment]
@danielroe I hope that might help, from a fresh nuxt project hosted on a CPanel VPS
### Environment
- Operating System: `Linux`
- Node Version:     `v16.15.0`
- Nuxt Version:     `3.0.0-rc.12`
- Nitro Version:    `0.6.0`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction
This project running script exists in ecosystem.config.js
[nuxt.zip](https://github.com/nuxt/framework/files/9840757/nuxt.zip)
pm2 command: ``pm2 start ecosystem.config.js``

### logs
![image](https://user-images.githubusercontent.com/58523735/197241915-ea70e44b-ee43-4efc-b7ec-0f7722696821.png)

**Comment by danielroe** [comment]
Do you get these same errors locally? (ie. not on the cpanel server)

**Comment by devhus** [comment]
Nope, It works locally but bugs in the production

**Comment by danielroe** [comment]
Is the environment info above from the cpanel environment or your local environment?

**Comment by danielroe** [comment]
How are you building and deploying your app? Do you have the configuration for that part of the piece?

**Comment by devhus** [comment]
> Is the environment info above from the cpanel environment or your local environment?

all info is from the cpanel environment nothing local, first env is for the main project and the second one is from a fresh nuxt project

**Comment by devhus** [comment]
> How are you building and deploying your app? Do you have the configuration for that part of the piece?

The app is building but never sees the light in production, After serving the project with PM2 instance in all ways it prints a 500-page error in the browser and logs the given logs above
Could you please explain to me what you mean by saying  "part of the piece" ? I've followed the deployment guide on nuxt's website and tried other methods to deploy the app using Supervisored plugin but it's not really good to work with

**Comment by danielroe** [comment]
This seems to be an issue with CPanel.

See [these search results](https://duckduckgo.com/?q=%22webassembly+instantiate+out+of+memory%22+cpanel).

Some users have found success upgrading their version of Node.

Note that [this comment](https://github.com/piscinajs/piscina/issues/158#issuecomment-1001163229) rightly observes this is not about _running out_ of memory but failing to allocate it...

**Comment by devhus** [comment]
So what should be done about that? Should i give it a try and upgrade my node version if needed?
Also, Note that the same server is used to host other Nuxt2 projects on exactly the same CPanel environment with no problem at all.

**Comment by devhus** [comment]
@danielroe So what should be done about that? Should i give it a try and upgrade my node version if needed?
Also, Note that the same server is used to host other Nuxt2 projects on exactly the same CPanel environment with no problem at all.

**Comment by danielroe** [comment]
As one issue observed, it may be triggered by having `type: module` in the `package.json`.

**Comment by devhus** [comment]
So basically is it not something that should be fixed in the library itself? Just need to know so i can start finding solutions 😞

**Comment by devhus** [comment]
@danielroe So basically is it not something that should be fixed in the library itself? Just need to know so i can start finding solutions 😞

**Comment by danielroe** [comment]
I _doubt_ it is an issue with Nuxt as exactly the same issue occurs with cpanel and Next.js and other runtime node code. I am happy to leave it open until I or someone else who uses cpanel can confirm that though.

**Comment by devhus** [comment]
Nothing new about this issue yet ?

**Comment by alfredogotay** [comment]
I have been running a node Express app in cPanel for over 6 months without issues but suddenly started getting the same error when trying to load a package (imgmin). 

`const imagemin = (await import("imagemin")).default  // this line throws the error`

The error occurs:
1. when running 'npm run start' from cPanel's terminal
2. when running as daemon with pm2 process manager

The error does not occur:
1. when running 'npm run start' from a terminal in my PC via SSH

I am running node version 16.6.1
